libcloud/compute/drivers/gce.py

```diff
@@ -2569,50 +2586,35 @@ def list_nodes(self, ex_zone=None, ex_use_disk_cache=True):
         :return:  List of Node objects
         :rtype:   ``list`` of :class:`Node`
         """
-        list_nodes = []
         zone = self._set_zone(ex_zone)
-        if zone is None:
-            request = '/aggregated/instances'
-        else:
-            request = '/zones/%s/instances' % (zone.name)
-        response = self.connection.request(request, method='GET').object
+        response = self.connection.request_aggregated_items('instances',
+                                                            zone=zone)
+
+        if not response.get('items', []):
```

Comment:
This return early change is purely for style purpose so we can make the code a bit easier to read and get rid of one level of indentation on "happy" code path when response contains results.